darwin/future/tests/core/test_query.py

```diff
@@ -47,9 +57,42 @@ def test_query_filter_functionality(basic_filters: List[Query.QueryFilter], test
     assert q.filters == basic_filters
 
 
-def test_query_iterable(basic_filters: List[Query.QueryFilter], test_team: Team) -> None:
-    q = Query.Query(test_team, basic_filters)
+def test_query_iterable(basic_filters: List[Query.QueryFilter], non_abc_query: Type[Query.Query]) -> None:
+    q = non_abc_query(basic_filters)
     for i, f in enumerate(q):
         assert f == basic_filters[i]
     assert q.filters is not None
     assert len(q) == len(basic_filters)
+
+
+@pytest.mark.parametrize(
+    "mod,param,check,expected",
```

Comment:
Have you seen hypothesis btw - testing assertions set that may be useful for this type of test.